crates/precompile/src/point_evaluation.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{Precompile, PrecompileAddress};
+use c_kzg::{Bytes32, Bytes48};
+use revm_primitives::{PrecompileResult, StandardPrecompileFn};
+
+pub const POINT_EVALUATION_PRECOMPILE: PrecompileAddress = PrecompileAddress(
+    crate::u64_to_b160(10),
+    Precompile::Standard(point_evaluation_run as StandardPrecompileFn),
+);
+use crate::Error;
+/// `BLS_MODULUS: = 52435875175126190479447740508185965837690552500527637822603658699938581184513`
+/// in big endian format
+const BLS_MODULUS: [u8; 32] = [
+    115, 237, 167, 83, 41, 157, 125, 72, 51, 57, 216, 8, 9, 161, 216, 5, 83, 189, 164, 2, 255, 254,
+    91, 254, 255, 255, 255, 255, 0, 0, 0, 1,
+];
+
+/// See: https://eips.ethereum.org/EIPS/eip-4844
+pub fn point_evaluation_run(input: &[u8], gas_limit: u64) -> PrecompileResult {
+    // The data is encoded as follows: versioned_hash | z | y | commitment | proof | with z and y being padded 32 byte big endian values
+    assert!(input.len() == 192);
```

Comment:
Is this good assumption?

Reply:
Perhaps not, Can we through an error if they aren't equal. Something like 
```rust
    if input.len() != 192 {
        return Err(Error::InvalidInputLength);
    }

```